lib/parser.mly

```diff
@@ -33,16 +33,15 @@
 %%
 
 program:
-  decls EOF { $1 }
+  decls { (fst $1, snd $1) }
 
-/* list of stmts and funcs */
 decls:
-   /* nothing */ { [] }
- | stmt decls    { (Stmt $1)::$2 }
- | fdecl decls   { (Func $1)::$2 }
+   /* nothing */ { ([], [])               }
+ | decls bind  { (($2 :: fst $1), snd $1) }
+ | decls fdecl { (fst $1, ($2 :: snd $1)) }
 
 bind:
-  dtype ID { ($1, $2) }
+  dtype ID EOL { ($1, $2) }
```

Comment:
I think this breaks the way we define functions since we use a list of binds separated by commas for the parameter list, but now it's looking for EOL between each bind.

Reply:
so true bestie